### PR TITLE
GPII-1308:  Update to use smtp-server...

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
In working on the tests for the new user management module, I noticed a few shortcomings in our mail test module that made it harder to work with the standard test framework.  This pull request addresses those concerns.

See [GPII-1308](https://issues.gpii.net/browse/GPII-1308) for details.

### CODE REVIEW

File: tests/lib/caseholder.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        // Confirm that the test content exists and is correct
+        var messageBody = fs.readFileSync(messageFile, "utf8");
+        jqUnit.assertTrue("There should be a message ID", messageBody.indexOf("Message-Id") !== -1);
+        jqUnit.assertTrue("The subject data should be in the message.", messageBody.indexOf(expected.subject) !== -1);
+        jqUnit.assertTrue("The message text should be in the message.", messageBody.indexOf(expected.text) !== -1);
+    }
+};
+
+// Copied from a similar utility grade in `gpii.express`.
+gpii.test.mail.caseholder.addRequiredSequences = function (sequenceStart, rawTests) {
```
Comment (amb26) [inline tests/lib/caseholder.js:35]: Can we now not use the similar utility from gpii-express?

Comment (the-t-in-rtf) [inline tests/lib/caseholder.js:35]: These are very different types of results under inspection, one is an HTTP response, the other is a mail session.  Concepts like status codes do not apply to the mail session, and other mail-isms do not apply to the HTTP response.

Comment (amb26) [inline tests/lib/caseholder.js:35]: This code is a duplicate of that at https://github.com/GPII/gpii-express/blob/master/tests/js/lib/test-helpers.js#L34 - this duplication should be eliminated

Comment (the-t-in-rtf) [inline tests/lib/caseholder.js:35]: Sorry, the comment appeared to be about the previous function.  I would rather not introduce gpii-express as a dev dependency just to pick this up, but I can do so.

Comment (amb26) [inline tests/lib/caseholder.js:35]: Perhaps a better approach would then be to factor out this piece of universally useful testing support as a "micromodule" of its own

Comment (the-t-in-rtf) [inline tests/lib/caseholder.js:35]: Yes, I was thinking we should make this part of the kettle test suite itself, and include a "delayed caseholder".

**Comment by the-t-in-rtf** [comment]
As I have updated the code surrounding the duplicated `addRequiredSequences`, I wanted to comment here so that it wasn't missed.  I am proposing submitting a wrapper grade as part of Kettle itself.  I am happy to prepare a pull request, or to have you do it, or to put it somewhere else.  I just need your input.

File: tests/all-tests.js [hunk 2]
```diff
@@ -1,92 +1,90 @@
...
+                            listener: "gpii.test.mail.caseholder.verifyMailBody",
+                            event: "{testEnvironment}.smtpServer.events.onMessageReceived",
+                            args: ["{testEnvironment}", "{that}.options.messages.basic"]
+                        },
+                        // This dummy check must be here or the mailer will be destroyed before its handler has been
+                        // allowed to do its job, which will result in a `fluid.fail` that breaks the test.
+                        //
+                        // TODO:  Review with Antranig
+                        {
+                            listener: "fluid.identity",
```
Comment (amb26) [inline tests/all-tests.js:64]: Yes, I think this is reasonable and necessary "for good reason".

**Pull request merged**